core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -0,0 +1,1045 @@
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/tevino/abool"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+const hibernationPollPeriod = 24 * time.Hour
+
+// Specification defines the Flux Monitor specification
+type Specification struct {
+	ID    string
+	JobID int32
+
+	ContractAddress     common.Address
+	Precision           int32
+	Threshold           float64
+	AbsoluteThreshold   float64
+	PollTimerPeriod     time.Duration
+	PollTimerDisabled   bool
+	IdleTimerPeriod     time.Duration
+	IdleTimerDisabled   bool
+	MinJobPayment       *assets.Link
+	TransmissionAddress common.Address // The address the Eth TX is sent from
+}
+
+// FluxMonitorFactory holds the New method needed to create a new instance
+// of a FluxMonitor.
+type FluxMonitorFactory interface {
+	New(Specification, PipelineRun, Config) (*FluxMonitor, error)
+}
+
+type fluxMonitorFactory struct {
+	store          Store
+	ethClient      eth.Client
+	logBroadcaster log.Broadcaster
+}
+
+// New constructs an instance of PollingDeviationChecker with sane defaults and
+// validation.
+func (f fluxMonitorFactory) New(
+	spec Specification,
+	pipelineRun PipelineRun,
+	cfg Config,
+) (*FluxMonitor, error) {
+	// Validate the poll timer
+	if !spec.PollTimerDisabled &&
+		spec.PollTimerPeriod < cfg.MinimumPollingInterval() {
+
+		return nil, fmt.Errorf(
+			"pollTimerPeriod must be equal or greater than %s",
+			cfg.MinimumPollingInterval(),
+		)
+	}
+
+	// Set up the flux aggregator
+	f.logBroadcaster.AddDependents(1)
+	fluxAggregator, err := flux_aggregator_wrapper.NewFluxAggregator(
+		spec.ContractAddress,
+		f.ethClient,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	contractSubmitter := NewFluxAggregatorContractSubmitter(fluxAggregator, spec.TransmissionAddress)
+
+	// Set up the contract flags
+	flags, err := NewFlags(cfg.FlagsContractAddress, f.ethClient)
+	logger.ErrorIf(
+		err,
+		fmt.Sprintf(
+			"unable to create Flags contract instance, check address: %s",
+			cfg.FlagsContractAddress,
+		),
+	)
+
+	paymentChecker := &PaymentChecker{
+		MinContractPayment: cfg.MinContractPayment,
+		MinJobPayment:      spec.MinJobPayment,
+	}
+
+	min, err := fluxAggregator.MinSubmissionValue(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	max, err := fluxAggregator.MaxSubmissionValue(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewFluxMonitor(
+		pipelineRun,
+		f.store,
+		NewPollTicker(spec.PollTimerPeriod, spec.PollTimerDisabled),
+		paymentChecker,
+		spec.ContractAddress,
+		contractSubmitter,
+		NewDeviationChecker(spec.Threshold, spec.AbsoluteThreshold),
+		*flags,
+		fluxAggregator,
+		f.logBroadcaster,
+		spec,
+		func() { f.logBroadcaster.DependentReady() },
+		min,
+		max,
+	)
+}
+
+// FluxMonitor polls external price adapters via HTTP to check for price swings.
+type FluxMonitor struct {
+	contractAddress   common.Address
+	oracleAddress     common.Address
+	pipelineRun       PipelineRun
+	store             Store
+	spec              Specification
```

Comment:
Ideally we can remove passing the spec to the FluxMonitor itself, but I haven't finished modularising the logic into separate component structs. Let me know if  I'm going too far here.

Reply:
I've removed the factory and made it into a constructor instead. The constructor takes a `job.SpecDB` instead of the `Specification` struct (this has been removed)